theano/gof/destroyhandler.py

```diff
@@ -891,6 +928,8 @@ def on_change_input(self, fgraph, app, i, old_r, new_r, reason):
 
                     self.view_o.setdefault(new_r, OrderedSet()).add(output)
 
+            if config.cycle_detection == 'fast':
```

Comment:
Instead of using a global config inside the op, I would put in the init of this class a new parameter: cycle_detection that default to None. When none, it use the config value. This would make the instance use an instance value instead of a global variable. This would allow user without changes to Theano to have different Theano function with different configuration of this variable.